### PR TITLE
Extract Connector IR from JAR

### PR DESCRIPTION
**Pull request by janelletavares**
## Description of change

Working with Flink Java to get an IR spec

Fixes https://github.com/meroxa/product/issues/854

## Type of change

- [x]  New feature
- [ ]  Bug fix
- [ ]  Refactor
- [ ]  Documentation

## How was this tested?

- [x]  Unit Tests
- [x]  Tested in staging
- [ ]  Tested in minikube

## Demo

| before | after |
|--------|-------|
|||

### CODE REVIEW

File: cmd/meroxa/flink/deploy.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		l.Warnf(ctx,
+			"warning: unable to verify local Java version is compatible with the Meroxa Platform; jar's must be compiled for %s",
+			exactJavaVersion)
+		return
+	}
+
+	// looks like 'openjdk version "11.0.19" 2023-04-18'
+	r := regexp.MustCompile(`version "([0-9.]+.[0-9.]+.[0-9.]+)"`)
```
Comment (samirketema) [inline cmd/meroxa/flink/deploy.go:82]: do we need to validate this _exact_ version? locally I have `openjdk64-11.0.18` installed, as well as the microsoft jdk 11

I wonder if we can generalize to just support java 11, though that is a bit more work

Comment (samirketema) [reply cmd/meroxa/flink/deploy.go:82]: Ah, I see that you're verifying `v11` in general. Though, it may not show in this exact format for the regex

Comment (samirketema) [reply cmd/meroxa/flink/deploy.go:82]: ok, I take this back, my local versions are matching this regex:
```
❯ java --version
openjdk 11.0.18 2023-01-17 LTS
OpenJDK Runtime Environment Microsoft-7208460 (build 11.0.18+10-LTS)
OpenJDK 64-Bit Server VM Microsoft-7208460 (build 11.0.18+10-LTS, mixed mode)
```

Comment (janelletavares) [reply cmd/meroxa/flink/deploy.go:82]: it's a bad constant name... I'll update to majorJavaVersion.

Comment (janelletavares) [reply cmd/meroxa/flink/deploy.go:82]: also, the output of -version and --version are slightly different :lolsob:

File: cmd/meroxa/flink/deploy.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = os.Stat(irFilepath)
+	if err != nil {
+		// @TODO try the docker way because the jar is skinny
+		// Otherwise, there are no Meroxa* classes in this main class
+		return nil, nil
+	}
```
Comment (samirketema) [inline cmd/meroxa/flink/deploy.go:53]: nit: could be an inline err

**Review by samirketema — APPROVED** [review]
this is great. will take another look if you push up more changes

**Pull request merged**